### ISSUE TITLE
rsync --chmod doesn't fully support chmod(1) mode syntax

### ISSUE THREAD

**Issue by steepnought** [WayneD/rsync#678]
According to the rsync 3.3.0 manpage, `rsync --chmod` should follow `the normal parsing rules specified in the chmod(1) manpage`. This is not quite true as the following doesn't work:

```
$ rsync -av --chmod=g=o,o= testfile1 testfile2
rsync: Invalid argument passed to --chmod (g=o)
rsync error: syntax or usage error (code 1) at main.c(1782) [client=3.3.0]
```

The intention here is to copy the permissions from others to group (`g=o`), then clear others' permissions (`o=`).

`chmod(1)` supports this (POSIX calls it permcopy).

**Comment by Kevin5025** [comment]
I was having the same issue but was able to do `--chmod=u=rw,--chmod=g=r,--chmod=o=r` as a workaround. 
I wonder if using multiple --chmod flags would solve your use case too.

**Comment by steepnought** [comment]
No, this is not the same issue. It's not about multiple chmod operations in one argument. I'm talking about copying permissions from one category to another which `chmod(1)` supports but rsync doesn't although the manpage says it should.

**Comment by steadytao** [comment]
Opened a small PR for this. It adds support for `u`, `g` and `o` permission copies on the right side of `--chmod` symbolic modes and includes a regression test for `g=o,o=`.

**Comment by steepnought** [comment]
Thanks for picking this up. But it's very sad that you weren't able to do it without "AI".

**Comment by steadytao** [comment]
It should be quickly noted that Andrew is retired 😅
